pkg/sys/mounter/mounter.go

```diff
@@ -17,18 +17,26 @@ limitations under the License.
 
 package mounter
 
-import (
-	"k8s.io/mount-utils"
-)
-
 const (
 	Binary = "/usr/bin/mount"
 )
 
-func NewMounter(binary string) mount.Interface {
-	return mount.NewWithoutSystemd(binary)
+type Interface interface {
+	Mount(source string, target string, fstype string, options []string) error
+	Unmount(target string) error
+	// IsMountPoint check /proc/mounts or equivalent data to check if the given path is listed there
+	IsMountPoint(path string) (bool, error)
+	// GetMountRefs finds all mount references to pathname, returning a slice of
+	// paths. The returned slice does not include the given path.
+	GetMountRefs(pathname string) ([]string, error)
+	// GetMountPoints parses /proc/mounts or equivalent data to fetch all available mountpoints for the given device
+	GetMountPoints(device string) ([]MountPoint, error)
 }
 
-func NewDummyMounter() mount.Interface {
-	return mount.NewFakeMounter([]mount.MountPoint{})
+// MountPoint represents a single line in /proc/mounts or /etc/fstab.
+type MountPoint struct {
```

Comment:
Since the interface interacts with this new type the interface definition had to be moved to a separate package to prevent circular dependencies. That's why there is  a nested package in mounter just for the implementation.

Reply:
This is no longer the case